src/components/lazy-list/index.js

```diff
@@ -0,0 +1,327 @@
+import { useState, useLayoutEffect, useRef, useMemo, useCallback } from 'react';
+
+/**
+ * A component that renders only the children currently visible on screen.
+ * @param {function} height A `function(start, end)` returning the pixel height for any given range of items
+ * @param {number} total The total count of all items in the list
+ * @param {function} children A `function(props)` rendering the list and items (see `childProps`)
+ * @param {?number} [buffer=0.5] A number [0...1] as a % of the visible region to render additionally
+ * @param {?boolean} [lazy=true] Toggles the lazy functionality
+ * @param {?boolean} [dispose=false] Toggles disposing items when they lose visibility
+ * @param {?function} onChange Optional change callback
+ * @param {?function} container Optional, default scroll container is `element.offsetParent`
+ * @return {object} The rendered children
+ **/
+export default ({
+  height,
+  total,
+  children,
+  lazy = true,
+  dispose = false,
+  buffer = 0.5,
+  onChange,
+  container = element => element?.offsetParent
+}) => {
+  // Required feature checks
+  const supported = typeof window.IntersectionObserver !== 'undefined';
+
+  // Active only if enabled by prop and features detected
+  const active = lazy && supported;
+
+  // The range of items currently rendered
+  const [range, setRange] = useState([0, 0]);
+  const rangeRef = useRef([0, 0]);
+
+  // List container element
+  const listRef = useRef();
+
+  // Upper placeholder element
+  const upperRef = useRef();
+
+  // Lower placeholder element
+  const lowerRef = useRef();
+
+  // Height of a single item
+  const itemHeight = useMemo(() => height(0, 1), [height]);
+
+  // Height of all items
+  const totalHeight = useMemo(() => height(0, total), [height, total]);
+
+  // Height of items above the rendered range
+  const upperHeight = useMemo(() => height(0, range[0]), [height, range]);
+
+  // Height of items below the rendered range
+  const lowerHeight = useMemo(() => height(range[1], total), [
+    height,
+    range,
+    total
+  ]);
+
+  // Skipped if not enabled or supported
+  if (active) {
+    // Allows an update only once per frame
+    const requestUpdate = useRequestFrameOnce(
+      // Memoise the frame callback
```

Comment:
small typo: should be memorize?

Reply:
I'm gonna jump in here and say no, because it's the verb form of memo, not memory: https://en.wikipedia.org/wiki/Memoization